### ISSUE TITLE
Sharing folders in a group folder does not work

### ISSUE THREAD

**Issue by Luto73** [nextcloud/groupfolders#1024]
I use the following structure for my group folders:

    Business1: write/share/delete access to ceo
        |
        |---Department1: write/share/delete access to _department1_ and _ceo_
        |
        |---Department2: write/share/delete access to _departement2_ and _ceo_
_ceo_ created a non-group-folder in the group folder "Business1" and shared it to _department1_. _department1_ can't access this shared folder.

Is there a way to fix this? Or do I have to create the folder within the admin interface of the group folder app and add the permissions I want?

I am using Nextcloud 18.0.7 and group folders 6.0.7

**Comment by pierreozoux** [comment]
We experience the same situation. 

Nextcloud 18.0.6 and  6.0.7.

GroupFolder write/share/delete access to group "groupName"

User1 from groupName share it with User2 that is not part of groupName.

User2 doesn't see anything about GroupFolder.

We use postgres and s3 if it helps debug.

**Comment by mamatt** [comment]
I can confirm this is also not working for us.
 - NC 18.0.4
 - GF 6.0.6

**Comment by vasyugan** [comment]
Same here. It used to work in previous versions. I can't say exactly, when it changed, but recently I am getting lots of complaints from users that they cannot see the contents of folders shared with them, and in each case, it turns out to be a subfolder of a group folder, and the only remedy is to manually delete the share in the database. Attempts to remove it through the web interface don't work.

**Comment by vasyugan** [comment]
The picture is inconsistent. In some cases it works, in others it does not. In some cases, deleting the share in the DB and then enabling extended permissions for myself and then re-creating the share has worked around this.

**Comment by ImaCrea** [comment]
Any news maybe around this topic please?

Or maybe some of you found a workaround?

**Comment by codello** [comment]
I also have this issue and I'm also using S3 storage as primary storage. I was searching around this issue and found nextcloud/server#24103 from a few days ago. I did a quick test with a default primary storage and was in fact able to create shares from inside group folders. Although I did not test that PR I have high hopes that it will resolve this issue…

**Comment by codello** [comment]
I just did a quick test with NextCloud version 19.0.6. Sharing items in group folders seems to be working for me now.

**Comment by pierreozoux** [comment]
@codello which commit did solve it then? I see the commit you mentionned only present in the version 21. Or did you apply the patch on 19.0.6 with success?

Thanks for your comments :)

**Comment by codello** [comment]
I'm going with my assumption that nextcloud/server#24103 is actually the pull request that solved the issue for me. In the discussion of that issue the changes were backported. The fix is probably available in the following versions (going by the milestones in the backport pull requests):
- Version 21: nextcloud/server#24103
- Version 20: nextcloud/server#24325 (version 20.0.3)
- Version 19: nextcloud/server#24327 (version 19.0.6)
- Version 18: nextcloud/server#24344 (version 18.0.12)

I only tested 19.0.6 though (using the Docker container) but I was able to successfully create shares from group folders. Maybe someone else can verify that 20.0.3 and 18.0.12 also resolve this issue.
